### PR TITLE
python311Packages.pyairvisual: drop asynctest

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Related https://github.com/NixOS/nixpkgs/pull/216168

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 216198` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.airvisual</li>
    <li>home-assistant-component-tests.airvisual_pro</li>
    <li>python310Packages.pyairvisual</li>
    <li>python311Packages.pyairvisual</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyairvisual/default.nix [hunk 1]
```diff
@@ -38,8 +37,11 @@ buildPythonPackage rec {
   ];
 
   nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  checkInputs = [
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyairvisual/default.nix:43]:
```suggestion
  nativeCheckInputs = [
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 216198` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.airvisual</li>
    <li>home-assistant-component-tests.airvisual_pro</li>
    <li>python310Packages.pyairvisual</li>
    <li>python311Packages.pyairvisual</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyairvisual/default.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/development/python-modules/pyairvisual/default.nix:43]: Time to ask the question that had been on my mind for a long time: should these *python* test deps in `nativeCheckInputs` or `checkInputs`? In some sense they are *linked* at test time, thus should be in `checkInputs`, just like why the runtime deps are in `buildInputs` (propagated), but the notion is just vague for python.

Comment (dotlambda) [reply pkgs/development/python-modules/pyairvisual/default.nix:43]: cc @mweinelt @FRidh @jonringer

Comment (mweinelt) [reply pkgs/development/python-modules/pyairvisual/default.nix:43]: Everything these days should go into `nativeBuildInputs`, at least all occurrences of `checkInputs` in our documentation were replaced like that.

Comment (mweinelt) [reply pkgs/development/python-modules/pyairvisual/default.nix:43]: https://github.com/NixOS/nixpkgs/blob/633891ec46d41b4461945fa7a06b2a329313850a/doc/stdenv/stdenv.chapter.md?plain=1#L142-L144

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 216198` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.airvisual</li>
    <li>home-assistant-component-tests.airvisual_pro</li>
    <li>python310Packages.pyairvisual</li>
    <li>python311Packages.pyairvisual</li>
  </ul>
</details>

**Pull request merged**